src/python/dxpy/dx_extract_utils/cohort_filter_payload.py

```diff
@@ -0,0 +1,141 @@
+    for compound_filter in filters["pheno_filters"]["compound"]:
+        if "entity" not in compound_filter or "name" not in compound_filter["entity"]:
+            continue
+        if compound_filter["entity"]["name"] != entity:
+            continue
+        if "logic" in compound_filter and compound_filter["logic"] != "and":
+            raise ValueError("Invalid input cohort. Cohorts must have 'and' logic on the primary entity and field.")
+        # The entity filter is valid for addition of field filter
+        compound_filter["filters"][entity_field] = [entity_field_filter]
+        return filters
+
+    # Search for an existing filter with the entity in a filter with a different field since no entity was found
+    for compound_filter in filters["pheno_filters"]["compound"]:
+        if "filters" not in compound_filter:
+            continue
+        for other_entity_field in compound_filter["filters"]:
+            if other_entity_field.split("$")[0] != entity:
+                continue
+            if "logic" in compound_filter and compound_filter["logic"] != "and":
+                continue
+            # Filter with the entity is valid for addition of field filter
+            compound_filter["filters"][entity_field] = [entity_field_filter]
+            return filters
+
+    # No existing filters for the entity were found so create the entity filter
+    filters["pheno_filters"]["compound"].append({
+        "name": "phenotype",
+        "logic": "and",
+        "filters": {
+            entity_field: [entity_field_filter],
+        },
+    })
+
+    return filters
+
+
+def cohort_filter_payload(values, entity, field, filters, project_context, base_sql=None):
+    if "logic" in filters and filters["logic"] != "and":
+        raise ValueError("Invalid input cohort. Cohorts must have 'and' logic on the primary entity and field.")
+    filter_payload = {}
+    filter_payload["filters"] = generate_pheno_filter(values, entity, field, filters)
+    if "logic" not in filter_payload["filters"]:
+        filter_payload["filters"]["logic"] = "and"
+    filter_payload["project_context"] = project_context
+    if base_sql is not None:
+        filter_payload["base_sql"] = base_sql
+
+    return filter_payload
+
+
+def cohort_combined_payload(combined):
+    combined = copy.copy(combined)
+    source = []
+    for source_dict in combined["source"]:
+        source.append({
+            "$dnanexus_link": {
+                "id": source_dict["id"],
+                "project": source_dict["project"],
+            }
+        })
+    combined["source"] = source
+
+    return combined
+
+
+def cohort_final_payload(name, folder, project, databases, dataset, filters, sql, base_sql=None, combined=None):
+    details = {
+        "databases": databases,
+        "dataset": {"$dnanexus_link": dataset},
+        "description": "",
+        "filters": filters,
+        "schema": "create_cohort_schema",
```

Comment:
?
Should this be schema copied from dataset details?

Reply:
Correct, will update.